pandas-stubs/core/indexes/base.pyi

```diff
@@ -626,6 +640,156 @@ class Index(IndexOpsMixin[S1]):
         self: Index[_str], other: _str | Sequence[_str] | np_ndarray_str | Index[_str]
     ) -> Index[_str]: ...
     @overload
+    def __sub__(self: Index[Never], other: DatetimeIndex) -> Never: ...
+    @overload
+    def __sub__(self: Index[Never], other: complex | _NumListLike | Index) -> Index: ...
+    @overload
+    def __sub__(self, other: Index[Never]) -> Index: ...  # type: ignore[overload-overlap]
+    @overload
+    def __sub__(
+        self: Index[bool],
+        other: Just[int] | Sequence[Just[int]] | np_ndarray_anyint | Index[int],
+    ) -> Index[int]: ...
+    @overload
+    def __sub__(
+        self: Index[bool],
+        other: Just[float] | Sequence[Just[float]] | np_ndarray_float | Index[float],
+    ) -> Index[float]: ...
+    @overload
+    def __sub__(
+        self: Index[int],
+        other: (
+            int
+            | Sequence[int]
+            | np_ndarray_bool
+            | np_ndarray_anyint
+            | Index[bool]
+            | Index[int]
+        ),
+    ) -> Index[int]: ...
+    @overload
+    def __sub__(
+        self: Index[int],
+        other: Just[float] | Sequence[Just[float]] | np_ndarray_float | Index[float],
+    ) -> Index[float]: ...
+    @overload
+    def __sub__(
+        self: Index[float],
+        other: (
+            float
+            | Sequence[float]
+            | np_ndarray_bool
+            | np_ndarray_anyint
+            | np_ndarray_float
+            | Index[bool]
+            | Index[int]
+            | Index[float]
+        ),
+    ) -> Index[float]: ...
+    @overload
+    def __sub__(
+        self: Index[complex],
+        other: (
+            T_COMPLEX
+            | Sequence[T_COMPLEX]
+            | np_ndarray_bool
+            | np_ndarray_anyint
+            | np_ndarray_float
+            | Index[T_COMPLEX]
+        ),
+    ) -> Index[complex]: ...
+    @overload
+    def __sub__(
+        self: Index[T_COMPLEX],
+        other: (
+            Just[complex]
+            | Sequence[Just[complex]]
+            | np_ndarray_complex
+            | Index[complex]
+        ),
+    ) -> Index[complex]: ...
+    @overload
+    def __sub__(
+        self: Index[Timestamp],
+        other: timedelta | np.timedelta64 | np_ndarray_td | TimedeltaIndex,
+    ) -> DatetimeIndex: ...
+    @overload
+    def __sub__(
+        self: Index[Timedelta],
+        other: timedelta | np.timedelta64 | np_ndarray_td | TimedeltaIndex,
+    ) -> TimedeltaIndex: ...
```

Comment:
Ideally, we wouldn't have a `Index[Timestamp]` or `Index[Timedelta]`, because those should be `DatetimeIndex` and `TimedeltaIndex`.  Could we move those 2 overloads to those 2 classes, respectively?